symphony-bdk-core-invokers/symphony-bdk-core-invoker-jersey2/src/main/java/com/symphony/bdk/core/api/invoker/jersey2/ApiClientBuilderJersey2.java

```diff
@@ -0,0 +1,180 @@
+package com.symphony.bdk.core.api.invoker.jersey2;
+
+import com.symphony.bdk.core.api.invoker.ApiClient;
+import com.symphony.bdk.core.api.invoker.ApiClientBuilder;
+
+import org.apiguardian.api.API;
+import org.glassfish.jersey.SslConfigurator;
+import org.glassfish.jersey.client.ClientConfig;
+import org.glassfish.jersey.client.ClientProperties;
+import org.glassfish.jersey.client.HttpUrlConnectorProvider;
+import org.glassfish.jersey.jackson.JacksonFeature;
+import org.glassfish.jersey.media.multipart.MultiPartFeature;
+
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+
+/**
+ * Specific implementation of {@link ApiClientBuilder} which creates a new instance of an {@link ApiClientJersey2}.
+ */
+@API(status = API.Status.STABLE)
+public class ApiClientBuilderJersey2 implements ApiClientBuilder {
+
+  private String basePath;
+  private byte[] keyStoreBytes;
+  private String keyStorePassword;
+  private byte[] trustStoreBytes;
+  private String trustStorePassword;
+  private Map<String, String> defaultHeaders;
+  private int connectionTimeout;
+  private int readTimeout;
+  private String temporaryFolderPath;
+
+  public ApiClientBuilderJersey2() {
+    basePath = "https://acme.symphony.com";
+    keyStoreBytes = null;
+    keyStorePassword = null;
+    trustStoreBytes = null;
+    trustStorePassword = null;
+    defaultHeaders = new HashMap<>();
+    connectionTimeout = 15000;
+    readTimeout = 60000;
+    temporaryFolderPath = null;
+    userAgent("Symphony BDK/" + getBdkVersion() + "/java/1.8");
+  }
+
+  /**
+   * Specific implementation of {@link ApiClientBuilder#buildClient()} which returns an {@link ApiClientJersey2} instance.
+   */
+  @Override
+  public ApiClient buildClient() {
+    java.util.logging.Logger.getLogger("org.glassfish.jersey.client").setLevel(java.util.logging.Level.SEVERE);
+
+    SSLContext sslContext = createSSLContext();
+
+    Client httpClient = ClientBuilder.newBuilder()
+        .sslContext(sslContext)
+        .withConfig(createClientConfig())
+        .build();
+
+    httpClient.property(ClientProperties.CONNECT_TIMEOUT, connectionTimeout);
+    httpClient.property(ClientProperties.READ_TIMEOUT, readTimeout);
+
+    return new ApiClientJersey2(httpClient, basePath, defaultHeaders, temporaryFolderPath);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ApiClientBuilder basePath(String basePath) {
```

Comment:
with this great pattern builder we should have all method with the same format name, I mean we should have all method names with the prefix with. withBastPath(String), withUserAgent, ....